src/drivers/stm32/drv_led_pwm.cpp

```diff
@@ -85,8 +85,9 @@
 #define rDMAR(_tmr)     REG(_tmr, STM32_GTIM_DMAR_OFFSET)
 #define rBDTR(_tmr)     REG(_tmr, STM32_ATIM_BDTR_OFFSET)
 
-static void             led_pwm_timer_init(unsigned timer);
-static void             led_pwm_timer_set_rate(unsigned timer, unsigned rate);
+
+extern int             io_timer_init_timer(unsigned timer);
+
```

Comment:
When I refactored the original pwm servo code at add input capture. I add the notion of io_timers that the input capture and new pwm servo are then layered on

The drv_led_pwm.cpp looks like a quick solution, by way of a copy of the pwm servo code, done by after the fact. 

Here are the 2 64$ questions: 1) Can the led_pwm code be refactored to do the RGB on top of the pwm servo or io_timer? 2) Would this and further refactoring create unwanted coupling?